File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        int best = Integer.MAX_VALUE;
        long sum = 0;
        int left = 0;

        for (int right = 0; right < sequence.length; right++) {
            sum += sequence[right];
            while (sum > k) {
                sum -= sequence[left++];
            }
            if (sum == k && right - left < best) {
                best = right - left;
                answer[0] = left;
                answer[1] = right;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = [0, 0]
    best = float('inf')
    total = 0
    left = 0

    for right in range(len(sequence)):
        total += sequence[right]
        while total > k:
            total -= sequence[left]
            left += 1
        if total == k and right - left < best:
            best = right - left
            answer[0] = left
            answer[1] = right
            
    return answer